saleor/graphql/decorators.py

```diff
@@ -72,3 +90,33 @@ def check_perms(context):
 staff_member_required = account_passes_test(
     lambda context: context.user.is_active and context.user.is_staff
 )
+
+
+def check_attribute_required_permissions():
+    """Check attribute permissions that depend on attribute type.
+
+    As an attribute can belong to the product or to the page,
+    different permissions need to be checked.
+    """
+
+    def check_perms(context, attribute):
+        if attribute.type == AttributeType.PAGE_TYPE:
+            return _permission_required((PagePermissions.MANAGE_PAGES,), context)
+        return _permission_required((ProductPermissions.MANAGE_PRODUCTS,), context)
+
+    return account_passes_test_for_attribute(check_perms)
+
+
+def check_attribute_value_required_permissions():
+    """Check attribute value permissions depending on the corresponding attribute type.
+
+    As an value's attribute can belong to the product or to the page,
+    different permissions need to be checked.
+    """
+
+    def check_perms(context, attribute_value):
+        if attribute_value.attribute.type == AttributeType.PAGE_TYPE:
```

Comment:
Will this perform a database query? Maybe it's not the issue for now but I'm wondering how this affects performance.

Reply:
I added benchmark tests for `attribute` and `attributes` query and it seems that adding this decorator doesn't change anything.

Reply:
👍